### PR TITLE
Fixed #35612 --Added note on reports of security vulnerability.

### PR DESCRIPTION
**Pull request by OkayJosh**
Users are responsibly or ensuring that their code adheres to the security standards.

# Trac ticket number

ticket-35612

# Branch description
Provide a concise overview of the issue or rationale behind the proposed changes.

# Checklist
- [ ] This PR targets the `main` branch. 
- [ ] The commit message is written in past tense, mentions the ticket number, and ends with a period.
- [ ] I have checked the "Has patch" ticket flag in the Trac system.
- [ ] I have added or updated relevant tests.
- [ ] I have added or updated relevant docs, including release notes if applicable.
- [ ] I have attached screenshots in both light and dark modes for any UI changes.

### CODE REVIEW

File: docs/internals/security.txt [hunk 1]
```diff
@@ -26,10 +26,20 @@ implications, please send a description of the issue via email to
...
+    ensure their code adheres to these standards, as Django cannot mitigate
+    vulnerabilities introduced by user errors (a common example being the failure
+    to sanitize user input), other examples can be found here :ref:`security
+    topics <cross-site-scripting>`.
+
 Once you've submitted an issue via email, you should receive an acknowledgment
 from a member of the security team within 48 hours, and depending on the
 action to be taken, you may receive further followup emails.
 
+
```
Comment (adamchainz) [inline docs/internals/security.txt:42]: undo, only one blank line needed
```suggestion
```

File: docs/internals/security.txt [hunk 2]
```diff
@@ -26,10 +26,20 @@ implications, please send a description of the issue via email to
...
 team <https://www.djangoproject.com/foundation/teams/#security-team>`_.
 
+.. note::
+
+    When reporting a potential security vulnerability, it is important that
+    your code follows security best practices. Users have a responsibility to
+    ensure their code adheres to these standards, as Django cannot mitigate
+    vulnerabilities introduced by user errors (a common example being the failure
+    to sanitize user input), other examples can be found here :ref:`security
+    topics <cross-site-scripting>`.
```
Comment (adamchainz) [inline docs/internals/security.txt:36]: Small grammatical tidy-ups - no need to parenthesize when a new sentence works well.

 ```suggestion
    When reporting a potential security vulnerability, it is important that
    your code follows security best practices. Users have a responsibility to
    ensure their code adheres to general standards, as Django cannot mitigate
    vulnerabilities introduced by user code. A common example is the failure
    to sanitize user input. For other examples, see the :ref:`security
    topics <cross-site-scripting>`.
```

**Comment by adamchainz** [comment]
“FIXED” should not be capitalized in your commit and PR titles, and it should end with a period. See examples: https://github.com/django/django/commits/main/

Thank you for adding this! I hope this will help limit bogus security reports, we get enough of them...

File: docs/internals/security.txt [hunk 3]
```diff
@@ -26,6 +26,15 @@ implications, please send a description of the issue via email to
...
 
+.. note::
+
+   When reporting a potential security vulnerability, it is important that
+   your code follows security best practices. Users have a responsibility to
+   ensure their code adheres to general standards, as Django cannot mitigate
+   vulnerabilities introduced by user code. A common example is the failure
+   to sanitize user input. For other examples, see the :ref:`security
+   topics <cross-site-scripting>`.
+
```
Comment (shangxiao) [inline docs/internals/security.txt:37]: Hm I had to read this a few times to understand it. Perhaps this suggestion is more comprehensible?

```suggestion
   Before reporting a potential security vulnerability, it is important to ensure that
   your own code follows security best practices, in order to allow maintainers the best course of action. Users have a responsibility to
   ensure their code adheres to general standards, as Django cannot mitigate
   vulnerabilities introduced by user code. A common example is the failure
   to sanitize user input. For other examples, see the :ref:`security
   topics <cross-site-scripting>`.
```

I'm not part of the security team so the "in order to allow maintainers the best course of action" part will need consideration for tweaking/removal - but the general gist here is to explicitly state the purpose of this paragraph 👍

Comment (adamchainz) [reply docs/internals/security.txt:37]: I like this edit, yes, it’s good to put in some reasoning. The only edit I’d make is to change “your own code” to “your code”, as “own” is redundant.

**Comment by adamchainz** [comment]
@OkayJosh can you squash and rebase your branch, fixing the commit title as you go? See https://adamj.eu/tech/2022/03/25/how-to-squash-and-rebase-a-git-branch/

**Comment by OkayJosh** [comment]
> @OkayJosh can you squash and rebase your branch, fixing the commit title as you go? See https://adamj.eu/tech/2022/03/25/how-to-squash-and-rebase-a-git-branch/

sure

File: docs/internals/security.txt (hunk 3 above)
Comment (sarahboyce) [reply docs/internals/security.txt]: How about something like the following, being just above the "Sending encrypted reports" admonition
```
.. admonition:: Example exploits must adhere to security standards

   Ensure your code follows security best practices before reporting a potential
   vulnerability or exploit. Developers have a responsibility to adhere to
   security standards, such as sanitizing user input. For other examples, see
   :doc:`security in Django </topics/security>`.

   Django will not mitigate vulnerabilities introduced by user code. That
   artificial intelligence can generate insecure code has no impact on developer
   responsibility, and hence requires no action from the security team.
```
I think we specifically wanted to call out code generated by ai as an argument of "even chatGPT is doing this" isn't a valid argument. I also updated "cannot" to "will not" as we chose to trust people will/should do things like sanitize inputs.

Comment (sarahboyce) [reply docs/internals/security.txt]: On the reasoning, I guess it is "to enable a smooth investigation" or something like this
Essentially, if a demonstration of a security exploit doesn't follow best practices it is mostly invalid. It _might_ not be invalid and just be a bad example - if that's the case, it's distracting from the "real" issue

Comment (shangxiao) [reply docs/internals/security.txt]: Yup 👍 

Just to be clear though about my main confusion was what point the paragraph was trying to make 😅 then I realised it was basically "make sure your own code is secure before submitting stuff" … keyword "before" (and I still maintain "own" isn't redundant as in not "your code that's using Django" rather "the code you yourself have written should be secure")

Comment (OkayJosh) [reply docs/internals/security.txt]: > How about something like the following, being just above the "Sending encrypted reports" admonition
> 
> ```
> .. admonition:: Example exploits must adhere to security standards
> 
>    Ensure your code follows security best practices before reporting a potential
>    vulnerability or exploit. Developers have a responsibility to adhere to
>    security standards, such as sanitizing user input. For other examples, see
>    :doc:`security in Django </topics/security>`.
> 
>    Django will not mitigate vulnerabilities introduced by user code. That
>    artificial intelligence can generate insecure code has no impact on developer
>    responsibility, and hence requires no action from the security team.
> ```
> 
> I think we specifically wanted to call out code generated by ai as an argument of "even chatGPT is doing this" isn't a valid argument. I also updated "cannot" to "will not" as we chose to trust people will/should do things like sanitize inputs.

i concor with adding it above the "Sending encrypted reports" admonition

**Comment by sarahboyce** [comment]
Superseded by: https://github.com/django/django/pull/18658

**Pull request closed without merging**